tests/common.c

```diff
@@ -515,6 +515,16 @@ static void test_str_replace(CuTest *tc) {
 	CuAssertIntEquals(tc, 19, n);
 	CuAssertStrEquals(tc, "<body text='black'>", str);
 
+	strcpy(str, "Apple");
+	n = str_replace("Apple", "Pineapple", &p);
```

Comment:
Now i'm actually checking this test, i don't see how this should result into `Pineapple`?

Reply:
My mistake. It should.